js/index.js

```diff
@@ -142,37 +137,46 @@ const createImageDiv = () => {
 }
 
 /* Settings Events */
-settingsButton.onclick = () => {
-  toggleElement(settingsForm)
-  if (settingsButton.innerText === "Show Settings") {
-    settingsButton.innerText = "Hide Settings"
-  } else if (settingsButton.innerText === "Hide Settings") {
-    settingsButton.innerText = "Show Settings"
-  }
-}
-
-qualitySelector.onchange = (event) => {
-  event.preventDefault();
-  quality = parseFloat(event.target.value, 10);
-}
-
-smoothingSelector.onchange = (event) => {
-  event.preventDefault();
-  smoothingOptions = event.target.value;
-}
-
-aspectRatioSelector.onchange = () => {
-  if (this.checked) aspectRatioPreserved = true;
-  else aspectRatioPreserved = false;
-}
-
-heightInput.onchange = (event) => {
-  inputHeight = parseFloat(event.target.value, 10);
-}
-
-widthInput.onchange = (event) => {
-  inputWidth = parseFloat(event.target.value, 10);
+const listen = (sel, event) => {
+  // only ever addEventListener to 'body' or 'form' and then check ev.target
+  document.body.addEventListener(event, (ev) => {
+    if (!ev.target.matches(sel)) {
+      return true;
+    }
+    else if (sel === 'button[name="js-settings-button"]' && event === 'click') {
```

Comment:
You're introducing more complexity than you need to. You could scope the action to the element so that the read doesn't have to read this whole block to see what a single interaction does:

```js
listen('button[name="js-settings-button"]', 'click', function () {
      toggleElement(settingsForm)
      if (settingsButton.innerText === "Show Settings") {
        settingsButton.innerText = "Hide Settings"
      } else if (settingsButton.innerText === "Hide Settings") {
        settingsButton.innerText = "Show Settings"
      }
})
```

or at the least you could be doing if { /*...*/ return } rather than having if else.

`if else` makes sense where it's not possible to return to exit - such as a look that must execute either piece of logic, but then must also execute additional logic.

In this case you have unrelated logic being grouped together.

Reply:
corrected typos ^^

Reply:
added examples ^^